### ISSUE TITLE
Scoreboard

### ISSUE THREAD

**Issue by jvyden** [LBPUnion/ProjectLighthouse#33]
Submitting:

```
[HTTP] 404, 0ms: POST /LITTLEBIGPLANETPS3_XML/scoreboard/user/2956?lbp2=true
[HTTP] <playRecord> <host>true</host> <type>7</type> <playerIds>jvyden</playerIds> <score>290</score> </playRecord> 
```

Fetching:

```
[HTTP] 404, 0ms: GET /LITTLEBIGPLANETPS3_XML/topscores/user/2976/1?pageStart=1&pageSize=1
```

**Comment by jvyden** [comment]
Upload now works in 6c65f5127227a004a758c65dbfc108402ee0e73a, just needs to be retrieved properly

**Comment by LumaLivy** [comment]
#37 covers retrieving top scores from community levels, with no support for friend filtering yet (different endpoint).

Needs work to support endpoints for Story levels:

`[HTTP] 404, 0ms: GET /LITTLEBIGPLANETPS3_XML/topscores/developer/152889/1?pageStart=1&pageSize=10`

**Comment by jvyden** [comment]
Merged #37 a while ago, and it works initially, but after completing a level the scores do not show. I wonder what the deal with that is :thinking:

**Comment by LumaLivy** [comment]
I had a few ideas so far regarding why the 2nd board won't show up, but unfortunately none of my solutions have worked so far.

It's important to note that on RPCS3 LBP2, the only requests logged by Lighthouse when a user steps on the scoreboard are:
```
[HTTP] 200, 88ms: GET /LITTLEBIGPLANETPS3_XML/friendscores/user/3/1
[HTTP] 200, 276ms: POST /LITTLEBIGPLANETPS3_XML/scoreboard/user/3?lbp2=true
```
I tried returning the new score data (surrounding the player's score) as the response to the `scoreboard` POST. Looks like this (in ScoreController.cs, `[HttpPost] SubmitScore`):

```
            ...
            else
            {
                this.database.Scores.Add(score);
            }
            await this.database.SaveChangesAsync();

            return this.Ok(await TopScores(score.SlotId, score.Type));
```

(I added some defaults to the query parameters so that if none are requested, it pulls the scores around the user's own score (because the ranking needs to be calculated inside `TopScores` in order to paginate to the user's score). That'll come in a PR eventually)

Anyways, that didn't work, and then I had the same problem with adding the `friendscores` endpoint, can't get those to show up either following the same format as `topscores/`.

If I had to guess, the game is missing some critical part of the XML in the response (and/or different element names compared to the pre-finish scoreboard that *does* show the top scores correctly). I figured it would have all been the same, but I don't see where else it would be retrieving the data.

I don't have a packet capture of a player completing a level by standing on a scoreboard. If anyone does, that may help a lot!
